terra-workspace-manager-jmeter/src/main/java/automation/jmeter/CreateWorkspaceSimulation.java

```diff
@@ -0,0 +1,307 @@
+        parametersTree.add(userParameters);
+        parametersTree.add(backendListener);
+
+        HashTree httpSamplerTree = new HashTree();
+        httpSamplerTree.add(httpSamplerProxy);
+        httpSamplerTree.add(httpSamplerProxy, parametersTree);
+
+        setUpThreadGroupTree.add(threadGroup, httpSamplerTree);
+
+        HashTree jsr223PreProcessorTree = new HashTree();
+        jsr223PreProcessorTree.add(jsr223PreProcessor);
+        setUpThreadGroupTree.add(setupThreadGroup, jsr223PreProcessorTree);
+
+        testPlanTree.add(testPlan, setUpThreadGroupTree);
+
+        return testPlanTree;
+    }
+
+    public static void main(String[] argv) throws Exception {
+
+        // for capturing HTTP traffic
+        // System.setProperty("http.proxyHost", "127.0.0.1");
+        // System.setProperty("http.proxyPort", "8888");
+        // for capturing HTTPS traffic
+        // System.setProperty("https.proxyHost", "127.0.0.1");
+        // System.setProperty("https.proxyPort", "8888");
+
+        File jmeterHome = new File("/apache-jmeter-5.3");
+        String slash = System.getProperty("file.separator");
+
+        File jmeterProperties = new File(jmeterHome.getPath() + slash + "bin" + slash + "jmeter.properties");
+
+        // JMeter initialization (properties, log levels, locale, etc)
+        JMeterUtils.setJMeterHome(jmeterHome.getPath());
+        JMeterUtils.loadJMeterProperties(jmeterProperties.getPath());
+        JMeterUtils.initLogging();// you can comment this line out to see extra log messages of i.e. DEBUG level
+        JMeterUtils.initLocale();
+
+        initializeTestPlan("Terra Workspace Manager Test Plan");
+        setUpLoopController();
+        setUpThreadGroup();
+        setJSR223PreProcessor();
+        setLoopController(1);
+        setThreadGroup(2, 2);
+        setHttpSampler("POST", "api/workspaces/v1");
+        setHeaders();
+        setUserParameters();
+        setupBackendListeners();
+        configureTestPlan();
+
+        // save generated test plan to JMeter's .jmx file format
+        SaveService.saveTree(testPlanTree, new FileOutputStream(jmeterHome + slash + "createworkspace.jmx"));
+
+        // add Summarizer output to get test progress in stdout like:
+        // summary = 2 in 1.3s = 1.5/s Avg: 631 Min: 290 Max: 973 Err: 0 (0.00%)
+        Summariser summer = null;
+        String summariserName = JMeterUtils.getPropDefault("summariser.name", "summary");
+        if (summariserName.length() > 0) {
+            summer = new Summariser(summariserName);
+        }
+
+        // Store execution results into a .jtl file
+        String logFile = jmeterHome + slash + "createworkspace.jtl";
+        ResultCollector logger = new ResultCollector(summer);
+        logger.setFilename(logFile);
+        testPlanTree.add(testPlanTree.getArray()[0], logger);
+
+        // Run Test Plan
+        //jMeterEngine.configure(testPlanTree);
+        //jMeterEngine.run();
```

Comment:
Nit: did you mean to leave these lines commented out?

Reply:
Yes these lines are commented out for now.